src/wp-includes/class-wp-customize-manager.php

```diff
@@ -2089,22 +2089,16 @@ public function remove_frameless_preview_messenger_channel() {
 		?>
 		<script>
 		( function() {
-			var urlParser, oldQueryParams, newQueryParams, i;
 			if ( parent !== window ) {
 				return;
 			}
-			urlParser = document.createElement( 'a' );
-			urlParser.href = location.href;
-			oldQueryParams = urlParser.search.substr( 1 ).split( /&/ );
-			newQueryParams = [];
-			for ( i = 0; i < oldQueryParams.length; i += 1 ) {
-				if ( ! /^customize_messenger_channel=/.test( oldQueryParams[ i ] ) ) {
-					newQueryParams.push( oldQueryParams[ i ] );
-				}
-			}
-			urlParser.search = newQueryParams.join( '&' );
-			if ( urlParser.search !== location.search ) {
-				location.replace( urlParser.href );
+			const url = new URL( location.href );
+			const queryParams = url.searchParams;
```

Comment:
In regards to anyone who may have concerns about the use of `const` here (which breaks in IE11), I'll note that frontend JS code from blocks now makes use of `const`, including the JS for the File, Image, Navigation, and Query blocks. A reminder that Internet Explorer is no longer [supported](https://make.wordpress.org/core/handbook/best-practices/browser-support/) by WordPress.